libraries/AP_UAVCAN/AP_UAVCAN.h

```diff
@@ -32,253 +37,151 @@
 #define UAVCAN_SRV_NUMBER 18
 #endif
 
-#define AP_UAVCAN_MAX_LISTENERS 4
-#define AP_UAVCAN_MAX_GPS_NODES 4
-#define AP_UAVCAN_MAX_MAG_NODES 4
-#define AP_UAVCAN_MAX_BARO_NODES 4
-#define AP_UAVCAN_MAX_BI_NUMBER 4
-
 #define AP_UAVCAN_SW_VERS_MAJOR 1
 #define AP_UAVCAN_SW_VERS_MINOR 0
 
 #define AP_UAVCAN_HW_VERS_MAJOR 1
 #define AP_UAVCAN_HW_VERS_MINOR 0
 
 #define AP_UAVCAN_MAX_LED_DEVICES 4
-#define AP_UAVCAN_LED_DELAY_MILLISECONDS 50
 
-class AP_UAVCAN {
+/*
+    Frontend Backend-Registry Binder: Whenever a message of said DataType_ from new node is received,
+    The Callback will invoke registery to register the node as separate backend.
+*/
+#define UC_REGISTRY_BINDER(ClassName_, DataType_) \
+class ClassName_ : public AP_UAVCAN::RegistryBinder<DataType_> { \
+    public: \
+        ClassName_() : RegistryBinder() {} \
+        ClassName_(AP_UAVCAN* uc,  void (*ffunc)(AP_UAVCAN*, uint8_t, const ClassName_&)): \
```

Comment:
They are not exactly same, Registry has one of the parameter as RegistryBinder instead of ClassName_. Again this method is meant as a workaround to move uavcan headers to source files. So one can forward declare ClassName_ in headers and bind it with uavcan message type inside source code.